### PR TITLE
Experimental scribble integration

### PR DESCRIPTION
**Pull request by dmuhs**
This is for property validation using MythX CLI and the in-house developed Scribble instrumentation tool. It's not public yet, but integration is already experimentally supported. That's how fast I am.

image

### CODE REVIEW

File: mythx_cli/analyze/command.py [hunk 1]
```diff
@@ -141,7 +170,7 @@ def analyze(
         create_group = analyze_config.get("create-group", False)
 
     mode = mode or analyze_config.get("mode") or "quick"
-    group_id = analyze_config.get("group-id") or group_id
+    group_id = analyze_config.get("group-id") or group_id or None
```
Comment (joaosantos15) [inline mythx_cli/analyze/command.py:173]: why was this added now as part of the scribble integration?

File: mythx_cli/analyze/command.py [hunk 2]
```diff
@@ -170,87 +204,45 @@ def analyze(
...
-    if not target:
-        if Path("truffle-config.js").exists() or Path("truffle.js").exists():
-            files, source_list = find_truffle_artifacts(Path.cwd())
-            if not files:
-                raise click.exceptions.UsageError(
-                    (
-                        "Could not find any truffle artifacts. Are you in the project root? "
-                        "Did you run truffle compile?"
-                    )
+    for analyze_mode, element in mode_list:
```
Comment (joaosantos15) [inline mythx_cli/analyze/command.py:209]: This is just nitpicking and probably not actionable, but `analyze_mode` might be confused with the analyses mode. Maybe something like `project_mode` would be better 🤔

File: mythx_cli/analyze/util.py [hunk 3]
```diff
@@ -1,16 +1,108 @@
...
+    """Detect Truffle projects in paths.
+
+    This function detects whether a Truffle project can be found
+    in the given project base path.
+
+    :param path: The path prefix to look in (e.g. the CLI target)
+    :param project_base: The truffle-specific path suffix
+    :return: Boolean indicating whether path contains a truffle project
+    """
+    return (path / "truffle-config.js").exists() or (path / "truffle.js").exists()
```
Comment (joaosantos15) [inline mythx_cli/analyze/util.py:33]: Should this also check if the project is well formatted? If the contracts are in the correct directory for instance?

File: mythx_cli/analyze/command.py (hunk 1 above)
Comment (dmuhs) [reply mythx_cli/analyze/command.py:173]: This is a simple fix on the side that makes the parameter definition more explicit. That way it's easier to see the state of `group_id` if neither a config file entry nor a CLI parameter are given. It's somewhat unrelated to scribble.

File: mythx_cli/analyze/command.py (hunk 2 above)
Comment (dmuhs) [reply mythx_cli/analyze/command.py]: Good catch! This is an old name from early development. I have renamed the tuple element `scenario` and the related enum holding the types `ScenarioModes` in 2530b53

File: mythx_cli/analyze/util.py (hunk 3 above)
Comment (dmuhs) [reply mythx_cli/analyze/util.py:33]: Checking that all contracts are in the right place would become fairly complex, e.g. when there are node modules and relative import paths involved. I consciously left this out here, because all we need to know is:
- Is the target a truffle project in the first place?
- Does it contain compiled artifacts?

The contract structure can be handled by truffle itself. If the project is valid, it will compile and we will see artifacts. If it's not, but a truffle config file exists, we throw an error later on. :+1:
